laravel/app/Http/Controllers/ScheduleDataController.php

```diff
@@ -581,270 +433,89 @@ private function insertOption()
-                        "attending" => $attending,
-                        "milestones" => $_REQUEST["milestones3"],
-                        "objectives" => $_REQUEST["objectives3"],
-                        "anesthesiologist_id" => $pref_anest3,
-                        "isValid" => 1,
-                    ]);
+                if (
+                    Option::where("date", $date)
+                        ->where("resident", $resident)
+                        ->where("option", $choice)
+                        ->count() != 0
+                ) {
+                    $notify = true;
+                    array_push($overwrittenChoices, $i + 1);
+
+                    Option::where("date", $date)
+                        ->where("resident", $resident)
+                        ->where("option", $choice)
+                        ->update([
+                            "schedule" => $split[$i],
+                            "attending" => $attending,
+                            "milestones" => $_REQUEST["milestones" . $choice],
+                            "objectives" => $_REQUEST["objectives" . $choice],
+                            "anesthesiologist_id" => $pref_anest[$i],
+                            "isValid" => 1,
+                        ]);
+                } else {
+                    // Insert data
+                    if (!is_null($schedule_data[$i])) {
+                        Option::insert([
+                            "date" => $date,
+                            "resident" => $resident,
+                            "schedule" => $split[$i],
+                            "attending" => $attending,
+                            "option" => $choice,
+                            "milestones" => $_REQUEST["milestones" . $choice],
+                            "objectives" => $_REQUEST["objectives" . $choice],
+                            "anesthesiologist_id" => $pref_anest[$i],
+                            "isValid" => 1,
+                        ]);
+                    }
+                }
             } else {
                 Option::where("date", $date)
                     ->where("resident", $resident)
                     ->where("option", $choice)
                     ->delete();
-                // ->update(['isValid'=>0]);
-            }
-        } else {
-            $overwrittenChoices[2] = 0;
-            if (!is_null($schedule_data3)) {
-                // Insert data
-                Option::insert([
-                    "date" => $date,
-                    "resident" => $resident,
-                    "schedule" => $split[2],
-                    "attending" => $attending,
-                    "option" => $choice,
-                    "milestones" => $_REQUEST["milestones3"],
-                    "objectives" => $_REQUEST["objectives3"],
-                    "anesthesiologist_id" => $pref_anest3,
-                    "isValid" => 1,
-                ]);
             }
         }
```

Comment:
It seems like the logic here has changed. I'm not sure that it was intentional.

Reply:
The logic was changed, but it should be for the better. If split has a 0 instead of a schedule_data id, then they didn't choose an OR for that choice. Since they didn't choose an OR, the option in the DB needs to be deleted for that choice. It does look like everything works per my testing.

Reply:
Sounds reasonable to me. @mmferguson do you mind taking a look at this too? Just want to be sure it's correct b/c it is technically a logic change.

Reply:
Logic looks good to me. Nice job Spencer!
I don't think it's necessary since `$split` will probably be renamed to `schedule_id` or something, but would it be worth it to have a comment that saying the if the schedule_id is 0, then there is no preference made for the option number `$i` and therefore the option needs to be deleted (it is now considered "None")?